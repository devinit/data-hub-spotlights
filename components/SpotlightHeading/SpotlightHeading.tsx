import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

interface SpotlightHeadingProps {
  className?: string;
}

const SpotlightHeading: FunctionComponent<SpotlightHeadingProps> = ({ className, children }) => {
  return <h2 className={classNames('spotlight__heading', className)}>{children}</h2>;
};

export { SpotlightHeading };
