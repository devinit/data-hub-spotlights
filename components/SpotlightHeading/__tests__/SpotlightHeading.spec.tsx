/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SpotlightHeading } from '../SpotlightHeading';

describe('SpotlightHeading', () => {
  test('renders correctly with a heading', () => {
    const renderer = TestRenderer.create(<SpotlightHeading>My Heading</SpotlightHeading>).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with the specificed class name', () => {
    const renderer = TestRenderer.create(<SpotlightHeading className="my-class" />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
