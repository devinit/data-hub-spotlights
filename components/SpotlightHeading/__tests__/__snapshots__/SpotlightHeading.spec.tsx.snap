// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightHeading renders correctly with a heading 1`] = `
<h2
  className="spotlight__heading"
>
  My Heading
</h2>
`;

exports[`SpotlightHeading renders with the specificed class name 1`] = `
<h2
  className="spotlight__heading my-class"
/>
`;
